server/app/controllers/MaintenanceController.scala

```diff
@@ -1,15 +1,74 @@
 package controllers
 
+import com.alexitc.playsonify.models.pagination.Limit
+import com.xsn.explorer.data.async.{BlockFutureDataHandler, TransactionFutureDataHandler}
+import com.xsn.explorer.gcs.{GolombCodedSet, GolombEncoding}
+import com.alexitc.playsonify.core.FutureOr.Implicits.FutureOps
+import com.xsn.explorer.data.anorm.AnormPostgresDataHandler
+import com.xsn.explorer.data.anorm.dao.BlockFilterPostgresDAO
+import com.xsn.explorer.executors.DatabaseExecutionContext
+import com.xsn.explorer.models.values.{Blockhash, Height}
 import controllers.common.{MyJsonController, MyJsonControllerComponents}
 import javax.inject.Inject
 import org.scalactic.Good
+import play.api.db.Database
 import play.api.libs.json.JsObject
 
 import scala.concurrent.Future
 
-class MaintenanceController @Inject()(components: MyJsonControllerComponents) extends MyJsonController(components) {
+class MaintenanceController @Inject()(
+    blockDataHandler: BlockFutureDataHandler,
+    transactionDataHandler: TransactionFutureDataHandler,
+    db: DBMigration,
+    components: MyJsonControllerComponents
+) extends MyJsonController(components) {
 
   def run(query: String) = public { _ =>
+    val startingHeight = query.toInt
+
+    logger.info(s"Migrating...")
+
+    (startingHeight to 0 by -1).foreach(height => {
+      logProgress(startingHeight, height)
```

Comment:
This approach tries to apply all filters in parallel which would cause misleading logs, you'll get a log when the item is visited while that doesn't mean that all previous blocks are migrated.